lib/semian/mysql2.rb

```diff
@@ -84,7 +99,7 @@ def connect(*args)
     def acquire_semian_resource(*)
       super
     rescue ::Mysql2::Error => error
-      if error.message =~ CONNECTION_ERROR
+      if error.message =~ CONNECTION_ERROR || error.class == PingFailure
```

Comment:
Related to the previous comment, if you don't make `PingFailure` a subclass, this condition will always be true.

Also I'd use `is_a?` since it's way more idiomatic.